include/dgl/runtime/ndarray.h

```diff
@@ -454,7 +464,19 @@ inline void NDArray::CopyFrom(DGLArray* other) {
 
 inline void NDArray::CopyFrom(const NDArray& other) {
   CHECK(other.data_ != nullptr);
-  CopyFrom(&(other.data_->dl_tensor));
+  // only one NDArray is pinned
+  // they are on diff devices
```

Comment:
Can you elaborate more? What are 'they'?

Reply:
Yeah I don't quite understand why we need a specific `RecordedCopyFromTo` under this particular condition.  Looks like this will be called when one array is pinned by PyTorch and the other is not, *and* one is on CPU and the other is on GPU.  Moreover, it doesn't seem to matter *which one* is on CPU or pinned by PyTorch, so I'm more confused.

Reply:
Yeah, the comment is not clear. Here, I want the if condition to check (1) it is a copy between two diff devices and (2) one of the array (cpu) is pinned by PyTorch. 
Given that the array on gpu can never be pinned (will raise an exception), i.e., `pinned_by_pyt_` always `false` if on gpu, the second constraint can be simplified to `data1.pinned_by_pyt_ != data2.pinned_by_pyt_`.

I agree it compromises the readability a lot. So I updated by (1) splitting the if condition (2) improving the comment.